evm/spec/framework.tex

```diff
@@ -33,5 +33,119 @@ \subsection{Field selection}
 
 \subsection{Cross-table lookups}
 \label{ctl}
+The various STARK tables carry out indepedent operations, but on shared values. We need to check that the shared values are identical in all the STARKs that require them. This is where cross-table lookups (CTLs) come in handy. 
 
-TODO
+Suppose STARK $S_1$ requires an operation -- say $Op$ -- that is carried out by another STARK $S_2$. Then $S_1$ writes the input and output of $Op$ in its own table, and provides the inputs to $S_2$. $S_2$ also writes the inputs and outputs in its rows, and the table's constraints check that $Op$ is carried out correctly. We then need to ensure that the inputs and outputs are the same in $S_1$ and $S_2$.
+
+In other words, we need to ensure that the rows -- reduced to the inputs and outputs -- of $S_1$ calling $Op$ are permutations of the rows of $S_2$ that carry out $Op$.
+
+To prove this, the first step is to only select the rows of interest in $S_1$ and $S_2$, and filter out the rest. Let $f_1$ be the filter for $S_1$ and $f_2$ the filter for $S_2$. $f_1$ and $f_2$ are constrained to be in $\{0, 1\}$. $f_1 = 1$ (resp. $f_2 = 1$) whenever the row at hand carries out $Op$ in $S_1$ (resp. in $S_2$), and 0 otherwise. Let also $(\alpha, \beta)$ be two challenges.
```

Comment:
```suggestion
To prove this, the first step is to only select the rows of interest in $S_1$ and $S_2$, and filter out the rest. Let $f_1$ be the filter for $S_1$ and $f_2$ the filter for $S_2$. $f_1$ and $f_2$ are constrained to be in $\{0, 1\}$. $f_1 = 1$ (resp. $f_2 = 1$) whenever the row at hand carries out $Op$ in $S_1$ (resp. in $S_2$), and 0 otherwise. Let also $(\alpha, \beta)$ be two random challenges.
```